src/render/python/integrator_v.cpp

```diff
@@ -216,7 +216,15 @@ MI_VARIANT class PyADIntegrator : public CppADIntegrator<Float, Spectrum> {
                             Sensor *sensor,
                             uint32_t seed = 0,
                             uint32_t spp = 0) override {
-        PYBIND11_OVERRIDE(TensorXf, Base, render_forward, scene, params, sensor, seed, spp);
+        py::gil_scoped_acquire gil;
+
+        py::function func = py::get_override(this, "render_forward");
+        if (func) {
+            auto o = func(scene, *((py::object *) params), sensor, seed, spp);
+            return o.template cast<TensorXf>();
+        }
+
+        return Base::render_forward(scene, params, sensor, seed, spp);
```

Comment:
I'm curious about this change. 

My guess is that the macro's behavior would not do the explicit casting and de-referencing of `params`. But then, what was happening?

Reply:
Without this explicit cast, on the Python side the `params` variable wouldn't be recognise as having the right type (e.g. `SceneParameters`).